notebooks/tetraopt_example_using_comms_channel.ipynb

```diff
@@ -0,0 +1,162 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "c6c9226f-0613-430a-abc8-73d8ef537ac2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import asyncio\n",
+    "from scipy import optimize\n",
+    "from tq42.organization import list_all as list_all_organizations\n",
+    "from tq42.project import list_all as list_all_projects\n",
+    "from tq42.experiment import list_all as list_all_experiments\n",
+    "from tq42.channel import Channel, Ask, Tell\n",
+    "from tq42.client import TQ42Client\n",
+    "from tq42.algorithm import AlgorithmProto\n",
+    "from tq42.compute import HardwareProto\n",
+    "import numpy as np\n",
+    "import OptimizationTestFunctions as otf\n",
+    "\n",
+    "from tq42.experiment_run import ExperimentRun\n",
+    "from tq42.algorithm import CvaOptParametersProto"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f6966322-e065-4edc-95b4-1fafc715031f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#the main function is declared as async because it contains several await expressions that require it to be asynchronous.\n",
+    "async def main():\n",
+    "    with TQ42Client() as client:\n",
+    "        client.login()\n",
```

Comment:
do we really always want to execute a login when the user executes this? Can't we just put a note please login before executing this thing. I always get annoyed when I reexecute this thing...